ionoscloud/resource_dbaas_pgsql_cluster.go

```diff
@@ -0,0 +1,391 @@
+						},
+					},
+				},
+			},
+			"synchronization_mode": {
+				Type:         schema.TypeString,
+				Description:  "Represents different modes of replication.",
+				Required:     true,
+				ValidateFunc: validation.All(validation.StringInSlice([]string{"ASYNCHRONOUS", "SYNCHRONOUS", "STRICTLY_SYNCHRONOUS"}, false)),
+			},
+			"from_backup": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Description: "The PostgreSQL version of your cluster.",
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"backup_id": {
+							Type:         schema.TypeString,
+							Description:  "The unique ID of the backup you want to restore.",
+							Required:     true,
+							ValidateFunc: validation.All(validation.StringIsNotWhiteSpace),
+						},
+						"recovery_target_time": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: " If this value is supplied as ISO 8601 timestamp, the backup will be replayed up until the given timestamp. If empty, the backup will be applied completely.",
+						},
+					},
+				},
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func resourceDbaasPgSqlClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(SdkBundle).DbaasClient
+
+	dbaasCluster, err := dbaasService.GetDbaasPgSqlClusterDataCreate(d)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	dbaasClusterResponse, _, err := client.CreateCluster(ctx, *dbaasCluster)
+
+	if err != nil {
+		diags := diag.FromErr(fmt.Errorf("an error occured while creating a dbaas cluster: %s", err))
+		return diags
+	}
+
+	d.SetId(*dbaasClusterResponse.Id)
+
+	for {
+		log.Printf("[INFO] Waiting for cluster %s to be ready...", d.Id())
+
+		clusterReady, rsErr := dbaasClusterReady(ctx, client, d)
+
+		if rsErr != nil {
+			diags := diag.FromErr(fmt.Errorf("error while checking readiness status of dbaas cluster %s: %s", d.Id(), rsErr))
+			return diags
+		}
+
+		if clusterReady {
+			log.Printf("[INFO] dbaas cluster ready: %s", d.Id())
+			break
+		}
+
+		select {
+		case <-time.After(SleepInterval):
+			log.Printf("[INFO] trying again ...")
+		case <-ctx.Done():
+			log.Printf("[INFO] create timed out")
+			diags := diag.FromErr(fmt.Errorf("dbaas cluster creation timed out! WARNING: your dbaas cluster (%s) will still probably be created after some time but the terraform state wont reflect that; check your Ionos Cloud account for updates", d.Id()))
+			return diags
+		}
+
+	}
+
+	return resourceDbaasPgSqlClusterRead(ctx, d, meta)
+}
+
+func resourceDbaasPgSqlClusterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	client := meta.(SdkBundle).DbaasClient
+
+	cluster, apiResponse, err := client.GetCluster(ctx, d.Id())
+
+	if err != nil {
+		if apiResponse != nil && apiResponse.Response != nil && apiResponse.StatusCode == 404 {
```

Comment:
maybe easier to use the httpNotFound function, to make sure we don't miss any api.Response nil checks